src/js/components/ProductManageMenu.js

```diff
@@ -0,0 +1,120 @@
+import { SELECTOR } from '../constants/selector.js';
+import { $ } from '../utils/dom.js';
+import { validateProductName, validateProductPrice, validateProductQuantity } from '../utils/validation.js';
+
+/* eslint-disable class-methods-use-this */
+export default class ProductManageMenu {
+  #state = {
+    products: [],
+  };
+
+  get state() {
+    return this.#state;
+  }
+
+  set state(state) {
+    this.#state = state;
+  }
+
+  init() {
+    this.#state = {
+      products: JSON.parse(localStorage.getItem('products')) ?? [],
+    };
+    this.#render();
+    this.#renderState();
+    this.#bindEvents();
+  }
+
+  #resetInput() {
+    $(SELECTOR.PRODUCT_NAME_INPUT).value = '';
+    $(SELECTOR.PRODUCT_PRICE_INPUT).value = '';
+    $(SELECTOR.PRODUCT_QUANTITY_INPUT).value = '';
+  }
+
+  #handleProductAddButtonClick() {
+    try {
+      const name = $(SELECTOR.PRODUCT_NAME_INPUT).value;
+      validateProductName(name);
+
+      const price = $(SELECTOR.PRODUCT_PRICE_INPUT).value;
+      validateProductPrice(price);
+
+      const quantity = $(SELECTOR.PRODUCT_QUANTITY_INPUT).value;
+      validateProductQuantity(quantity);
+
+      const product = { name, price, quantity };
+
+      this.#addProduct(product);
+      this.#resetInput();
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  #addProduct(product) {
+    const targetIndex = this.#state.products.findIndex((stateProduct) => stateProduct.name === product.name);
+    const isNotExist = targetIndex === -1;
```

Comment:
`targetIndex`라는 변수로 분리하는데 크게 의미가 있지는 않을 것 같아서 두 줄로 분리하는 이득이 딱히 없어보여요 🤔 
```suggestion
    const currentProductNotStored = this.#state.products.findIndex((stateProduct) => stateProduct.name === product.name) === -1;
```

Reply:
코드를 다시 보니까 저도 똑같은 느낌이 드는 것 같습니다.
오히려 한 줄로 처리하는 게 훨씬 괜찮아 보입니다!